Installation Issue: bowtie build error

### Steps to reproduce the issue

```console
[centos] ~: spack install bowtie
==> Installing bowtie
==> Searching for binary cache of bowtie
==> Warning: No Spack mirrors are currently configured
==> No binary for bowtie found: installing from source
==> Fetching https://github.com/BenLangmead/bowtie/archive/v1.2.2_p1.tar.gz
######################################################################## 100.0%
==> Staging archive: /spack/var/spack/stage/bowtie-1.2.2_p1-se66bd5p6mfiop65vwqpr4jh6uwvpxsr/v1.2.2_p1.tar.gz
==> Created stage in /spack/var/spack/stage/bowtie-1.2.2_p1-se66bd5p6mfiop65vwqpr4jh6uwvpxsr
==> No patches needed for bowtie
==> Building bowtie [MakefilePackage]
==> Executing phase: 'edit'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j2' 'NO_TBB=1'

4 errors found in build log:
     18    In file included from sequence_io.h:12:0,
     19                     from multikey_qsort.h:8,
     20                     from diff_sample.h:13,
     21                     from blockwise_sa.h:19,
     22                     from ebwt.h:27,
     23                     from ebwt_build.cpp:11:
  >> 24    pat.h:6:18: fatal error: zlib.h: No such file or directory
     25     #include <zlib.h>
     26                      ^

     27    compilation terminated.
     28    In file included from sequence_io.h:12:0,
     29                     from multikey_qsort.h:8,
     30                     from diff_sample.h:13,
     31                     from blockwise_sa.h:19,
     32                     from ebwt.h:27,
     33                     from ebwt_build.cpp:11:
  >> 34    pat.h:6:18: fatal error: zlib.h: No such file or directory
     35     #include <zlib.h>
     36                      ^
     37    compilation terminated.
  >> 38    make: *** [bowtie-build-l] Error 1
     39    make: *** Waiting for unfinished jobs....
  >> 40    make: *** [bowtie-build-s] Error 1

See build log for details:
  /spack/var/spack/stage/bowtie-1.2.2_p1-se66bd5p6mfiop65vwqpr4jh6uwvpxsr/bowtie-1.2.2_p1/spack-build.out
```

### Platform and user environment

Please report your OS here:
```commandline
$ uname -a
Linux 4b5226354c71 4.9.125-linuxkit #1 SMP Fri Sep 7 08:20:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
``` 
Bowtie installation fails with missing zlib dependency.

--- a/var/spack/repos/builtin/packages/bowtie/package.py
+++ b/var/spack/repos/builtin/packages/bowtie/package.py
@@ -38,6 +38,7 @@ class Bowtie(MakefilePackage):
     variant('tbb', default=False, description='Use Intel thread building block')
 
     depends_on('tbb', when='+tbb')
+    depends_on('zlib')
 
     # See: https://github.com/BenLangmead/bowtie/issues/87, a
     # different fix is in the FreeBSD ports/package tree

